### ISSUE TITLE
Inconsistent default values for null equality and NaN equality

### ISSUE THREAD

**Issue by bdice** [rapidsai/cudf#11329]
In `cudf::distinct`, we default to [**equal nulls and equal NaNs**](https://github.com/rapidsai/cudf/blob/719f4c8399f54340f51aeb37c79035dfbc3cc749/cpp/include/cudf/stream_compaction.hpp#L279-L280). This aligns with the defaults of `intersect_distinct`, `union_distinct`, and `difference_distinct` in PR #11043 (as of this writing - not yet merged).

However, there are competing standards:
- In `make_collect_set_aggregation`, we default to [**equal nulls and unequal NaNs**](https://github.com/rapidsai/cudf/blob/719f4c8399f54340f51aeb37c79035dfbc3cc749/cpp/include/cudf/aggregation.hpp#L519-L520).
- In `drop_list_duplicates`, we default to [**equal nulls and unequal NaNs**](https://github.com/rapidsai/cudf/blob/719f4c8399f54340f51aeb37c79035dfbc3cc749/cpp/include/cudf/lists/drop_list_duplicates.hpp#L81-L82).

We should strive for a consistent set of defaults for how null and NaN equality are handled. I think I would recommend a change in `make_collect_set_aggregation` and `drop_list_duplicates` to align with the other functions mentioned above.

_Originally posted by @bdice in https://github.com/rapidsai/cudf/pull/11043#discussion_r927186236_

**Comment by bdice** [comment]
I caught another place where `nan_equality::UNEQUAL` is being used in `make_merge_sets_aggregation`. https://github.com/rapidsai/cudf/blob/536ddd08fd2e86940c55dab178f4a1a1e9361539/cpp/include/cudf/aggregation.hpp#L592

I will make a follow-up PR that can be merged after #11621.

**Comment by ttnghia** [comment]
How about renaming `nan_equality::ALL_EQUAL` to just `EQUAL` to match with `null_equality`?

**Comment by vyasr** [comment]
Not quite done yet. @bdice be careful with the wording on your PRs :) GH sees "partially resolves X" as ".... resolves X" and still closes the issue AFAICT
